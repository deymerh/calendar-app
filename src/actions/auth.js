import Swal from 'sweetalert2';
import { fetchWithoutToken, fetchWithToken } from '../helpers/fetch';
import { types } from '../types/types';

export const startLogin = (email, password) => {
  return async (dispatch) => {
    try {
      const res = await fetchWithoutToken('auth', { email, password }, 'POST');
      const body = await res.json();
      if (body.ok) {
        localStorage.setItem('token', body.token);
        localStorage.setItem('token-init-date', new Date().getTime());
        dispatch(login({
          uid: body.uid,
          name: body.name
        }));
      } else {
        Swal.fire('Error', body.msg, 'error')
      }
    } catch (error) {
      console.log(error);
    }
  }
}
export const startRegister = (name, email, password) => {
  return async (dispatch) => {
    try {
      const res = await fetchWithoutToken('auth/register', { name, email, password }, 'POST');
      const body = await res.json();
      if (body.ok) {
        localStorage.setItem('token', body.token);
        localStorage.setItem('token-init-date', new Date().getTime());
        dispatch(login({
          uid: body.uid,
          name: body.name
        }));
        Swal.fire('Ok', body.msg, 'success');
      } else {
        Swal.fire('Error', body.msg, 'error');
      }
    } catch (error) {
      console.log(error);
    }
  }
}
export const startChecking = () => {
  return async (dispatch) => {
    const res = await fetchWithToken('auth/renew');
    const body = await res.json();
    if (body.ok) {
      localStorage.setItem('token', body.token);
      localStorage.setItem('token-init-date', new Date().getTime());
      dispatch(login({
        uid: body.uid,
        name: body.name
      }));
    } else {
      dispatch(checkingSinish());
    }
  }
}

export const startLogout = () => {
  return (dispatch) => {
    localStorage.clear();
    dispatch(logout)
  }
}
const logout = () => ({ type: types.authLogout });

const login = (user) => ({
  type: types.authLogin,
  payload: user
});

const checkingSinish = () => ({ type: types.authCheckingFinish });