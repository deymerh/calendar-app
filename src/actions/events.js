import Swal from 'sweetalert2';
import { fetchWithToken } from '../helpers/fetch';
import { prepareEvents } from '../helpers/prepareEvents';
import { types } from '../types/types';

export const eventStartAddNew = (event) => {
  return async (dispatch, getState) => {
    const { name, uid } = getState().auth;
    try {
      const res = await fetchWithToken('events', event, 'POST');
      const body = await res.json();
      if (body.ok) {
        event.id = body.event.id;
        event.user = {
          _id: uid,
          name: name
        }
        dispatch(eventAddNew(event));
      }
    } catch (error) {
      console.log(error);
    }
  };
};

const eventAddNew = (event) => ({
  type: types.eventAddNew,
  payload: event
});

export const eventSetActive = (event) => ({
  type: types.eventSetActive,
  payload: event
});

export const eventClearActive = () => ({ type: types.eventClearActive });

export const eventStartDeleted = () => {
  return async (dispatch, getState) => {
    const { id } = getState().calendar.eventActive;
    try {
      const res = await fetchWithToken(`events/${id}`, {}, 'DELETE');
      const body = await res.json();
      if (body.ok) {
        dispatch(eventDeleted());
        Swal.fire('Ok', body.msg, 'success');
      } else {
        Swal.fire('Error', body.msg, 'error');
      }
    } catch (error) {
      console.log(error);
    }
  };
};

const eventDeleted = () => ({ type: types.eventDeleted });

export const startEventsLoaded = () => {
  return async (dispatch) => {
    const res = await fetchWithToken('events');
    const body = await res.json();
    const events = prepareEvents(body.events);
    dispatch(eventsLoaded(events));
  };
};

export const eventStartUpdate = (event) => {
  return async (dispath) => {
    try {
      const res = await fetchWithToken(`events/${event.id}`, event, 'PUT');
      const body = await res.json();
      console.log(body);
      if (body.ok) {
        dispath(eventUpdated(event));
      } else {
        Swal.fire('Error', body.msg, 'error');
      }
    } catch (error) {
      console.log(error);
    }
  }
}

const eventUpdated = (event) => ({
  type: types.eventUpdated,
  payload: event
});

const eventsLoaded = (events) => ({
  type: types.eventsLoaded,
  payload: events
});

export const eventLogout = () => ({ type: types.eventsLogout });