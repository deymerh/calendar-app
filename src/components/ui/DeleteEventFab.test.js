import React from 'react';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import { DeleteEventFab } from './DeleteEventFab';
import { Provider } from 'react-redux';
import moment from 'moment';
import { eventStartDeleted } from '../../actions/events';

jest.mock('../../actions/events', () => ({
  eventStartDeleted: jest.fn()
}))

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initialState = {
  events: [
    {
      id: new Date().getTime(),
      title: 'Mi cumpleaÃ±os',
      start: moment().toDate(),
      end: moment().add(2, 'hours').toDate(),
      bgColor: '#fafafa',
      user: {
        _id: '12345',
        name: 'Deymer'
      }
    }
  ],
  eventActive: null
}

let store = mockStore(initialState);
store.dispatch = jest.fn();

describe('Test in the file [DeleteEventFab]', () => {

  beforeEach(() => {
    store = mockStore(initialState);
  });

  test('should do Snapshot the component [<DeleteEventFab />]', async () => {
    const { component } = render(
      <Provider store={store}>
        <DeleteEventFab />
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });

  test('should do Snapshot the component [<DeleteEventFab />]', async () => {
    store.dispatch = jest.fn();
    const { findByTestId } = render(
      <Provider store={store}>
        <DeleteEventFab />
      </Provider>
    );
    const btnDelete = await findByTestId('btnDelete');
    fireEvent.click(btnDelete);
    expect(eventStartDeleted).toHaveBeenCalled()

  });
});