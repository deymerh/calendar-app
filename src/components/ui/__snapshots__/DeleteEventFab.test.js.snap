// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in the file [DeleteEventFab] should do Snapshot the component [<DeleteEventFab />] 1`] = `undefined`;
